File: src/test/java/application/browser/PhaseHelper.java
package application.browser;

import static com.matag.player.PlayerType.OPPONENT;
import static com.matag.player.PlayerType.PLAYER;
import static org.openqa.selenium.support.ui.ExpectedConditions.attributeContains;
import static org.openqa.selenium.support.ui.ExpectedConditions.textToBe;
import static org.openqa.selenium.support.ui.ExpectedConditions.visibilityOfElementLocated;

import org.openqa.selenium.By;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import com.matag.player.PlayerType;

public class PhaseHelper {
  public static final By PHASE_CSS_SELECTOR = By.cssSelector("#turn-phases .active");
  public static final By PRIORITY_CSS_SELECTOR = By.cssSelector("#turn-phases .active");

  private static final Logger LOGGER = LoggerFactory.getLogger(PhaseHelper.class);

  private MatagBrowser matagBrowser;

  PhaseHelper(MatagBrowser matagBrowser) {
    this.matagBrowser = matagBrowser;
  }

  public void is(String phase, PlayerType priority) {
    matagBrowser.wait(textToBe(PHASE_CSS_SELECTOR, phase));
    isPriority(priority);
  }

  public void isPriority(PlayerType priority) {
    try {
      matagBrowser.wait(attributeContains(PRIORITY_CSS_SELECTOR, "class", getClassNameLinkedToPriority(priority)));
    } catch (Exception e) {
      LOGGER.warn("PhaseAndPriority:{} {}", getPhase(), getPriority());
      throw e;
    }
  }

  public String getPhase() {
    matagBrowser.wait(visibilityOfElementLocated(PHASE_CSS_SELECTOR));
    return matagBrowser.findElement(PHASE_CSS_SELECTOR).getText();
  }

  public PlayerType getPriority() {
    matagBrowser.wait(visibilityOfElementLocated(PRIORITY_CSS_SELECTOR));
    var classes = matagBrowser.findElement(PRIORITY_CSS_SELECTOR).getAttribute("class");
    if (classes.contains("active-for-player")) {
      return PLAYER;
    } else {
      return OPPONENT;
    }
  }

  private String getClassNameLinkedToPriority(PlayerType priority) {
    if (priority == PLAYER) {
      return "active-for-player";
    } else {
      return "active-for-opponent";
    }
  }
}

File: src/test/kotlin/application/browser/PhaseHelper.kt
package application.browser

import com.matag.player.PlayerType
import org.openqa.selenium.By
import org.openqa.selenium.support.ui.ExpectedConditions
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class PhaseHelper internal constructor(private val matagBrowser: MatagBrowser) {
    fun `is`(phase: String, priority: PlayerType?) {
        matagBrowser.wait(ExpectedConditions.textToBe(PHASE_CSS_SELECTOR, phase))
        isPriority(priority)
    }

    fun isPriority(priority: PlayerType?) {
        try {
            matagBrowser.wait(
                ExpectedConditions.attributeContains(
                    PRIORITY_CSS_SELECTOR,
                    "class",
                    getClassNameLinkedToPriority(priority)
                )
            )
        } catch (e: Exception) {
            LOGGER.warn("PhaseAndPriority:{} {}", this.phase, this.priority)
            throw e
        }
    }

    val phase: String
        get() {
            matagBrowser.wait(ExpectedConditions.visibilityOfElementLocated(PHASE_CSS_SELECTOR))
            return matagBrowser.findElement(PHASE_CSS_SELECTOR).getText()
        }

    val priority: PlayerType
        get() {
            matagBrowser.wait(ExpectedConditions.visibilityOfElementLocated(PRIORITY_CSS_SELECTOR))
            val classes =
                matagBrowser.findElement(PRIORITY_CSS_SELECTOR).getAttribute("class")
            if (classes!!.contains("active-for-player")) {
                return PlayerType.PLAYER
            } else {
                return PlayerType.OPPONENT
            }
        }

    private fun getClassNameLinkedToPriority(priority: PlayerType?): String {
        if (priority == PlayerType.PLAYER) {
            return "active-for-player"
        } else {
            return "active-for-opponent"
        }
    }

    companion object {
        val PHASE_CSS_SELECTOR: By = By.cssSelector("#turn-phases .active")
        val PRIORITY_CSS_SELECTOR: By = By.cssSelector("#turn-phases .active")

        private val LOGGER: Logger = LoggerFactory.getLogger(PhaseHelper::class.java)
    }
}
